Fix HVAC modes for Airzone slave zones

Noltari opened the pull request:
## Proposed change
HVAC modes must be updated on every coordinator update since HVAC mode should only be changed on the master zone and that change will be reflected on every slave zone after that.
This bug was introduced during the review of the Climate platform PR: https://github.com/home-assistant/core/pull/67924.

Release notes:
- https://github.com/Noltari/aioairzone/releases/tag/0.2.2
- https://github.com/Noltari/aioairzone/releases/tag/0.2.3

Git compare: https://github.com/Noltari/aioairzone/compare/0.2.1...0.2.3

## Type of change
- [x] Dependency upgrade https://github.com/Noltari/aioairzone/compare/0.2.1...0.2.3
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/core/pull/68798
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Noltari committed:
airzone: fix slave HVAC modes updates

Slave HVAC modes aren't being updated when HVAC mode is changed on the master
zone.
This regression was introduced during the review of the climate platform.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
I'll close this for now since I need to test this deeper...

Noltari committed:
airzone: update aioairzone to v0.2.2

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
> I'll close this for now since I need to test this deeper...

Should be fixed now... It required a change of the library:
https://github.com/Noltari/aioairzone/compare/0.2.1...0.2.2

bdraco commented on homeassistant/components/airzone/climate.py:
```diff
@@ -184,4 +181,7 @@ def _async_update_attrs(self) -> None:
         else:
             self._attr_hvac_action = CURRENT_HVAC_OFF
             self._attr_hvac_mode = HVAC_MODE_OFF
+        self._attr_hvac_modes = [
+            HVAC_MODE_LIB_TO_HASS[mode] for mode in self.get_zone_value(AZD_MODES)
+        ]
```
We don't allow `_attr_hvac_modes` to change after the start event since it will break anything that looks at the value (ie Voice Assistants such as Aleax, Google Assistant, HomeKit, etc)

bdraco replied:
Instead set all the supported modes the device can support and raise HomeAssistantError if they choose a mode that isn't available  in the current state

Noltari replied:
Ok, I will do it that way then

Noltari replied:
@bdraco introduced the suggested changes on 692bd223f02979ef17d57f01704bbad60f1f44ee

Noltari replied:
And I forgot to revert the HVAC modes change, but it should be solved in c1753c0a859210581ff1823acadb50955258a655.
Sorry for that!

Noltari committed:
airzone: fix HVAC modes on slave zones

With these changes the full list of HVAC modes is displayed on each slave zone,
but the HVAC mode can only be changed on the master zone. Therefore, the slave
zones can only be turned off or on by selecting the current master zone mode.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
Now the full list of HVAC modes is displayed on each slave zone, but it can only be changed on the master zone.
An example of what happens when you try to change slave zone Despacho from HVAC_MODE_HEAT to HVAC_MODE_COOL:
<img width="411" alt="image" src="https://user-images.githubusercontent.com/264346/160772071-8ee52a9b-728d-424f-982b-7ea2e9b0fc43.png">

Noltari committed:
tests: airzone: fix climate tests

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
airzone: revert _attr_hvac_modes change

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

bdraco commented on tests/components/airzone/test_climate.py:
```diff
@@ -179,6 +191,43 @@ async def test_airzone_climate_set_hvac_mode(hass):
+        "homeassistant.components.airzone.AirzoneLocalApi.http_request",
+        return_value=HVAC_MOCK,
+    ):
+        try:
```
use `with pytest.raises(HomeAssistantError)` instead

Noltari replied:
Changed in 5421593f1e8fd25795cc93f64017acd8abdd7682

Noltari committed:
tests: airzone: climate: use pytest.raises

Also fix some indents.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented on tests/components/airzone/test_climate.py:
```diff
@@ -231,8 +280,9 @@ async def test_airzone_climate_set_temp_error(hass):
                 },
                 blocking=True,
             )
+            assert False
```
Changed in 5421593f1e8fd25795cc93f64017acd8abdd7682
